src/include/sof/math/fir_config.h

```diff
@@ -44,7 +44,7 @@
 #if XCHAL_HAVE_HIFI2EP == 1
 #define FIR_HIFIEP	1
 #define FIR_HIFI3	0
-#elif XCHAL_HAVE_HIFI3 == 1
+#elif XCHAL_HAVE_HIFI3 == 1 || XCHAL_HAVE_HIFI4 == 1
```

Comment:
Good catch! I didn't realize that HIFI4 build does not have HIFI3 set while they are mostly compatible. I  wonder if there's other similar issues in code?

Reply:
actually in our config, we enabled HIFI3 while HIFI4 build. Add || XCHAL_HAVE_HIFI4 == 1 is because I'm not sure if the HIFI3 will still enabled while we have HIFI5 or other newer version(but might HIFI4 is enabled). I can  check the other code.